### DISCUSSION TITLE
[Design] Do we need a CDN API?

### DISCUSSION THREAD

**Discussion by KCarretto** [spellshift/realm#22] [Ideas]
Based on the discussion in [Design Deployment](https://github.com/KCarretto/realm/discussions/18) #18, Tomes will be able to contain all of their dependencies and therefore not need to download files from a CDN. This makes the CDN API seem a bit redundant, since most Tomes will just bundle in their dependencies. However, a potential use-case for the CDN API is for deploying large files to many targets at the same time. Although it will be possible to do this by bundling files into Tomes, it may be more performant in some cases for agents to download files from the CDN. Additionally, supporting a CDN API for file upload/downloads means that other tooling (that lacks support for Tomes) would be able to more easily use Realm as necessary. In the past, we've used the CDN API for simple scripts which we curl | bash. 

Do these use-cases justify supporting the CDN API?

**Comment by Cictrone** [comment]
I’m neutral. I think the pro of deprecation here is less UI complexity and less code. The cons will be we don’t offer a CDN/links solution which I think has been used in the past (but don’t know how much recently @hulto 👀)

**Reply by KCarretto** [reply]
That's a fair point, let's see what @hulto thinks

**Reply by hulto** [reply]
From a unified red team perspective I think we'll need it.
Most red teamers we'll work with will expect one for sharing files, and scripts. For both deployment and as a central location for file sharing.
Not a high priority. PWNdrop currently fills this need. The main gets are centralization, user integration, and hopefully an improved workflow managing/sharing tomes and tome components.
